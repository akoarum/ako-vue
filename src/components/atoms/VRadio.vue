<template>
  <label class="vRadio">
    <input
      v-model="model"
      v-bind="$attrs"
      :value="value"
      type="radio"
      class="vRadio__input"
    />
    <span class="vRadio__label"><slot /></span>
  </label>
</template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  inheritAttrs: false,
  model: {
    prop: 'checked',
    event: 'change'
  },
  props: {
    value: { type: String, required: true },
    checked: { type: String, required: true }
  },
  computed: {
    model: {
      get(): string {
        return this.checked
      },
      set(newValue: string) {
        this.$emit('change', newValue)
      }
    }
  }
})
</script>

<style lang="scss" scoped>
.vRadio {
  position: relative;
  padding-left: 26px;
  font-size: $fontSize-pc;
}

.vRadio__input {
  position: absolute;
  left: 0;
  top: 0;
  width: 10px;
  height: 10px;
  opacity: 0;
}

.vRadio__input:checked + .vRadio__label::before {
  content: 'radio_button_checked';
  color: $color-theme;
}

.vRadio__label::before {
  position: absolute;
  left: 0;
  top: -0.03em;
  content: 'radio_button_unchecked';
  font-size: 2.4rem;
  font-weight: normal;
  line-height: 1;
  font-family: $fontFamily-icons;
  color: #888;
  transition: color 0.1s $easing;
}
</style>
