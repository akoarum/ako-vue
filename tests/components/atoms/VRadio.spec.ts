import { shallowMount } from '@vue/test-utils'
import VRadio from '~/components/atoms/VRadio.vue'

describe('VRadio', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = shallowMount(VRadio, {
      propsData: {
        value: 'test',
        checked: ''
      },
      slots: {
        default: 'テスト'
      }
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
