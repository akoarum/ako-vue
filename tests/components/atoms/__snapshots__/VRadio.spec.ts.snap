// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VRadio initialized snapshot 1`] = `
<label
  class="vRadio"
>
  <input
    class="vRadio__input"
    type="radio"
    value="test"
  />
   
  <span
    class="vRadio__label"
  >
    テスト
  </span>
</label>
`;
